crud/options.go

```diff
@@ -0,0 +1,134 @@
+package crud
+
+import (
+	"errors"
+	"reflect"
+
+	"github.com/markphelps/optional"
+)
+
+type option interface {
+	getInterface() (interface{}, error)
+}
+
+type OptUint struct {
```

Comment:
What do we need this wrapper for? To hide the implementation?

Reply:
Yes, it needed to hide implementation and not to bother user to import `optional` package:

```golang
import (
	"github.com/markphelps/optional"
)

var selectOpts = crud.SelectOpts{
	Timeout: optional.NewUint(timeout),
}
```